gems/aws-sdk-core/lib/aws-sdk-core/rest/request/body.rb

```diff
@@ -17,7 +17,12 @@ def initialize(serializer_class, rules)
         # @param [Seahorse::Client::Http::Request] http_req
         # @param [Hash] params
         def apply(http_req, params)
-          http_req.body = build_body(params)
+          body = build_body(params)
+          # for rest-json, ensure we send at least an empty object for
+          if body.nil? && @serializer_class == Json::Builder && @rules[:payload]
+            body = Json.dump({})
```

Comment:
We can just hard code the empty body string (avoid extra serialization overhead).  Though I'd also argue this should be in the `build_body` method instead of here as we're already checking on whether there are params and payload.

Reply:
Agree on hard coded body string. I don't think it belongs in `build_body` because it applies in all three cases I think. The body must default to `{}` when modeled with the `payload` trait or when there are structures on the body (may not be marked with `payload`).